upload/upload.go

```diff
@@ -93,6 +96,41 @@ func loadConfigFile(path string) (*Config, error) {
 	return config, nil
 }
 
+// Function to determine whether the token is expiring in less than a day
+func checkTokenExpiration(accessToken string) (bool, error) {
+
+	// Parse jwt token with unverifies, since we don't need to check the signatures here
+	token, _, err := new(jwt.Parser).ParseUnverified(accessToken, jwt.MapClaims{})
+	if err != nil {
+		return false, fmt.Errorf("could not parse token, reason: %s", err)
+	}
+
+	// Return error if token is broken (without claims)
+	if claims, ok := token.Claims.(jwt.MapClaims); !ok {
+		return false, fmt.Errorf("broken token (claims are empty): %v\nerror: %s", claims, err)
+	}
+
+	var expiration time.Time
+	if claims, ok := token.Claims.(jwt.MapClaims); ok {
+		// Check if the token has exp claim
+		if claims["exp"] == nil {
+			return false, fmt.Errorf("could not parse token, reason: no expiration date")
+		}
+		switch iat := claims["exp"].(type) {
+		case float64:
+			expiration = time.Unix(int64(iat), 0)
+		case json.Number:
+			tmp, _ := iat.Int64()
+			expiration = time.Unix(tmp, 0)
+		}
+
+	}
+
```

Comment:
These 2 could have been concatenated into an if-else clause since they check the same thing.

Reply:
You are right, fixed now